crates/typst/src/layout/grid/rowspans.rs

```diff
@@ -696,6 +789,69 @@ impl<'a> GridLayouter<'a> {
             // Height of the latest spanned gutter row.
             // Zero if it was removed.
             let mut latest_spanned_gutter_height = Abs::zero();
+
+            let mut header_height = self.header_height;
+
+            let simulate_header_layout =
```

Comment:
Looking at these closures, I wonder whether the simulator should be a struct with methods in itself.

Reply:
Yeah that sounds more appropriate. I will investigate doing this refactor; if it ends up taking too long, we can delay it to the next PR.

Reply:
Done.